src/common/logging.cpp

```diff
@@ -54,12 +54,20 @@ static const char   sLevelString[][8] = {
       "[EMERG]", "[ALERT]", "[CRIT]", "[ERR ]", "[WARN]", "[NOTE]", "[INFO]", "[DEBG]",
 };
 
+static otbrLogLevel sDefaultLevel = OTBR_LOG_INFO;
+
```

Comment:
I want to echo same things as other reviewers mentioned about the `sDefaultLevel` and how it is used.
I feel it may be better to remove it and logic will be smipler:
- If the `aFeatureFlagList.enable_detailed_logging()` is used, then we control log level.
- If it is not set, we let the log level be whatever specified during build time.
  - OT core (and I think otbr) provide cmake `OT_LOG_LEVEL` option to specify this
  - I think this config may be set differently when otbr is integrated into products

Reply:
The `sDefaultLevel` is introduced to handle the corner case where `aFeatureFlagList.enable_detailed_logging()` is changed from `true` to `false` at run time. In this case we may want to set log level back to what was used in the beginning of otbr start.
When otbr is integrated into products the compile time log level can be override by otbr command line arguments at run time (https://github.com/openthread/ot-br-posix/blob/main/src/agent/main.cpp#L213), so I set `sDefaultLevel` to be same as otbr's final choice at otbrLogInit: https://github.com/openthread/ot-br-posix/pull/1678/files#diff-ee208097f13a01b500e564c1b15f89e2184eaeb0b61f71f7bcbe684bc75709f8R88

Reply:
Got it. Thanks. Sounds good.